lib/react_client.dart

```diff
@@ -135,69 +162,371 @@ class ReactDartComponentFactoryProxy<TComponent extends Component> extends React
+  }
+}
+
+/// Creates ReactJS [ReactElement] instances for DOM components.
+class ReactDomComponentFactoryProxy extends ReactComponentFactoryProxy {
+  /// The name of the proxied DOM component.
+  ///
+  /// E.g. `'div'`, `'a'`, `'h1'`
+  final String name;
+
+  /// The JS component factory used by this factory to build [ReactElement]s.
+  ///
+  /// Deprecated: Use [React.createElement()] instead and pass in [type] as
+  /// the first argument, followed by `props` and `children`.
+  ///
+  /// Before:
+  /// ```
+  ///   YourFactoryProxy.reactComponentFactory(props, children);
+  /// ```
+  ///
+  /// After:
+  /// ```
+  ///   React.createElement(YourFactoryProxy.type, props, children);
+  @Deprecated('6.0.0')
+  final Function factory;
+
+  ReactDomComponentFactoryProxy(name)
+      : this.name = name,
+        this.factory = React.createFactory(name) {
+    // TODO: Should we remove this once we validate that the bug is gone in Dart 2 DDC?
+    if (ddc_emulated_function_name_bug.isBugPresent) {
+      ddc_emulated_function_name_bug.patchName(this);
+    }
+  }
+
+  @override
+  String get type => name;
+
+  @override
+  ReactElement build(Map props, [List childrenArgs = const []]) {
+    var children = _convertArgsToChildren(childrenArgs);
+    children = listifyChildren(children);
+
+    // We can't mutate the original since we can't be certain that the value of the
+    // the converted event handler will be compatible with the Map's type parameters.
+    var convertibleProps = new Map.from(props);
+    convertProps(convertibleProps);
+
+    // jsifyAndAllowInterop also handles converting props with nested Map/List structures, like `style`
+    return React.createElement(type, jsifyAndAllowInterop(convertibleProps), children);
+  }
+
+  /// Performs special handling of certain props for consumption by ReactJS DOM components.
+  static void convertProps(Map props) {
+    _convertEventHandlers(props);
+    _convertRefValue(props);
+  }
+}
+
+/// Create react-dart registered component for the HTML [Element].
+_reactDom(String name) {
+  return new ReactDomComponentFactoryProxy(name);
+}
+
+void _convertRefValue(Map args) {
+  var ref = args['ref'];
+  if (ref is Ref) {
+    args['ref'] = ref.jsRef;
+  }
+}
+
+void _convertRefValue2(Map args, {bool convertCallbackRefValue = true}) {
+  var ref = args['ref'];
+
+  if (ref is Ref) {
+    args['ref'] = ref.jsRef;
+  } else if (ref is _CallbackRef && convertCallbackRefValue) {
+    args['ref'] = allowInterop((dynamic instance) {
+      if (instance is ReactComponent && instance.dartComponent != null) return ref(instance.dartComponent);
+      return ref(instance);
+    });
+  }
+}
+
+class ReactJsContextComponentFactoryProxy extends ReactJsComponentFactoryProxy {
+  /// The JS class used by this factory.
+  @override
+  final ReactClass type;
+  final bool isConsumer;
+  final bool isProvider;
+
+  /// Deprecated: Use [React.createElement()] instead and pass in [type] as
```

Comment:
The first line is usually a description of what the API is. Could we explicitly add that line and then add the deprecation below it?